check_diff/src/lib.rs

```diff
@@ -4,8 +4,26 @@ use std::path::Path;
 use std::process::Command;
 use tracing::info;
 
+pub enum CheckDiffError {
+    FailedGit(GitError),
+    FailedCommand(String),
+    FailedUtf8(String),
+    FailedSourceBuild(String),
+    FailedCopy(String),
+    IO(std::io::Error),
+}
+
+impl From<io::Error> for CheckDiffError {
+    fn from(error: io::Error) -> Self {
+        CheckDiffError::IO(error)
+    }
+}
```

Comment:
To improve the ergonomics you should add a conversion from `GitError` -> `CheckDiffError`.

```rust
impl From<GitError> for CheckDiffError {
    fn from(error: GitError) -> Self {
        CheckDiffError::FailedGit(error)
    }
}
```